src/tox/venv.py

```diff
@@ -515,6 +515,11 @@ def _pcall(
         env = self._get_os_environ(is_test_command=is_test_command)
         bin_dir = str(self.envconfig.envbindir)
         env["PATH"] = os.pathsep.join([bin_dir, os.environ["PATH"]])
+        if "PATH" in self.envconfig.setenv:
+            env["PATH"] = os.pathsep.join([env["PATH"], self.envconfig.setenv["PATH"]])
+            # Remove redundantly duplicated directories in PATH
+            env["PATH"] = ':'.join(list(set(env["PATH"].split(":"))))
```

Comment:
I don't think we should do that. A set is unordered and `PATH` is ordered - if I put myself in the shoes of the user, I would be surprised if that order would be arbitrarily changed. So I would completely remove that line.